apps/04TowersOfHanoi.js

```diff
@@ -13,30 +13,91 @@ var stacks = {
   c: []
 };
 
+var totalMoves=0;
+var minimalMoves = Math.pow(2, stacks['a'].length) - 1;
+console.log("Minimal moves is "+minimalMoves)
+
 function printStacks() {
   console.log("a: " + stacks.a);
   console.log("b: " + stacks.b);
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
+function movePiece(startStack,endStack) {
   // Your code here
+  //Remove from the first stack.
+  
+  var moveChar=stacks[startStack].pop();
+  //Place on the ending stack.
+  stacks[endStack].push(moveChar)
 
+  totalMoves++;
 }
 
-function isLegal() {
+function isLegal(startStack,endStack) {
   // Your code here
+  if ((startStack==='a' || startStack==='b' || startStack==='c') && (endStack==='a' || endStack==='b' || endStack==='c'))
+    {
+    //I have a valid input of a b or c.
+    var startStackLength=stacks[startStack].length;
+    var endStackLength=stacks[endStack].length;
+    if (startStackLength>0)
+      {
+      //There's stuff in this stack, so it is a valid starting stack.
+      var startStackChar=stacks[startStack][(startStackLength-1)];
```

Comment:
For readability, please include spaces when using logical/comparison operators or assigning stuff to variables.

ex:  `var winString = 'HOLY CRAP PLAYER ' + playerTurn + ' WON!!!!!'`